sharkfuser/build_tools/cmake/CTestMacros.cmake

```diff
@@ -82,10 +139,84 @@ function(add_sharkfuser_sample)
     return()
   endif()
 
-  _add_sharkfuser_target(
+  _add_sharkfuser_test(
     NAME ${_RULE_NAME}
     SRCS ${_RULE_SRCS}
     DEPS ${_RULE_DEPS}
     BIN_SUBDIR samples
   )
 endfunction()
+
+
+# Creates a lit test that compiles a source file and runs lit on it.
+#
+#  add_sharkfuser_lit_test(
+#    SRC <source-file>
+#    [TOOLS <tool1> <tool2> ...]
+#    [DEPS <dep1> <dep2> ...]
+#  )
+#
+# SRC
+#  The source file to compile and test (required)
+#
+# TOOLS
+#  External tools needed for the test (e.g., FileCheck)
+#
+# DEPS
+#  Library dependencies for the compiled executable
+function(add_sharkfuser_lit_test)
+  if(NOT SHARKFUSER_BUILD_TESTS)
+    return()
+  endif()
+
+  cmake_parse_arguments(
+    _RULE               # prefix
+    ""                  # options
+    "SRC"               # one value keywords
+    "TOOLS;DEPS"        # multi-value keywords
+    ${ARGN}
+  )
```

Comment:
minor: if early return works here, could we change the other functions to do the same for consistency? For some reason I thought the `cmake_parse_arguments` needs to happen first, maybe I was wrong.